config_operator/config_operator/config_operator.py

```diff
@@ -0,0 +1,39 @@
+import argparse
+from config_operator.config_source import RemoteGitConfig, LocalDirConfig
+from config_operator.k8s import K8sConfigMap
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Run git configuration synchronization operator, work on local-dir or git-url")
+    input_group = parser.add_mutually_exclusive_group(required=True)
+    input_group.add_argument("-l", "--local-dir",
+                             help="local directory where the configuration files are")
+    input_group.add_argument("-gu", "--git-url",
+                             help="git repository from which the configuration files are pulled/saved")
+    parser.add_argument("-gb", "--git-branch", help="git branch from which the configuration files are pulled/saved",
+                        default="master")
+    parser.add_argument("-gl", "--git-local", help="local git repository", required=False)
+    parser.add_argument("-gt", "--git-token", help="git personal access token used to access the repository")
+    parser.add_argument("-n", "--namespace", help="kubernetes namespace where the configuration will be deployed", required=True)
+    parser.add_argument("-cm", "--config-map", help="configmap name in kubernetes", required=True)
```

Comment:
Can this option be made optional and have a default value assigned?

Reply:
I am not sure about that. If the default value is optional then it has not be known by the collection_manager (the default value will be hidden inside the python code and not visible from k8s or helm). We can discuss that. This is not difficult to do though.

With the operator, the configuration looks like:
git-repo-test.yml:
apiVersion: sdap.apache.org/v1
kind: gitBasedConfig
metadata:
  name: gitbasedconfig
spec:
  git-url: "https://github.com/tloubrieu-jpl/sdap-ingester-config"
  config-map: "collection-ingester-conf"
  update-every-seconds: 10

It is deployed with:
kubectl apply -f containers/k8s/git-repo-test.yml -n sdap

Reply:
Okay you're right, the config-map name should not be hidden inside the python code since it needs to be coordinated with the collection manager. Anyway, looks like this is taken care of with the operator config now.